### PR TITLE
Change error message in LockWithTimeout and write UT for the same

### PR DESCRIPTION
**Pull request by vlaxman-px**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: etcd/v3/kv_etcd.go [hunk 1]
```diff
@@ -733,7 +734,7 @@ func (et *etcdKV) LockWithTimeout(
 			}
 		}
 		if err != nil && time.Since(startTime) > lockTryDuration {
-			return nil, err
+			return nil, errors.New("Lock already taken")
```
Comment (adityadani) [inline etcd/v3/kv_etcd.go:737]: nit: all golang errors should start with a lower case. Lets also change the error message to:
```
fmt.Errorf("failed to take a lock on %v: lock taken by: %v ", key, currLockerTag.LockerID)
```

File: test/kv.go [hunk 2]
```diff
@@ -883,8 +884,43 @@ func lockBetweenRestarts(kv kvdb.Kvdb, t *testing.T, start StartKvdb, stop StopK
...
+	time.Sleep(2 * time.Second)
+	go getTimeoutLock(kv, key, "timeouttest2", tryDuration, holdDuration, kvChan, errChan)
+	kvPair1, kvPair2 := <-kvChan, <-kvChan
+	err1, err2 := <-errChan, <-errChan
+	assert.NotNil(t, kvPair1, "Problem in acquiring Lock")
+	assert.Nil(t, kvPair2, "Lock acquired incorrectly")
+	assert.NoError(t, err1, "Unexpected error while trying to acquire lock")
+	assert.Error(t, err2, "Unexpected behaviour")
+	// error return while failing to acquire lock
+	fmt.Println(err2)
```
Comment (adityadani) [inline test/kv.go:911]: No need to print the error here

File: test/kv.go [hunk 3]
```diff
@@ -883,8 +884,43 @@ func lockBetweenRestarts(kv kvdb.Kvdb, t *testing.T, start StartKvdb, stop StopK
...
+	assert.NoError(t, err, "Unexpected error while unlock")
+	go getTimeoutLock(kv, key, "timeouttest3", tryDuration, holdDuration, kvChan, errChan)
+	kvPair2 = <-kvChan
+	assert.NotNil(t, kvPair2, "Problem in acquiring Lock after Unlock")
+
+	fmt.Println("lockTimeoutError() ended successfully")
+}
 
+func lockBasic(kv kvdb.Kvdb, t *testing.T) {
+	fmt.Println("Starting lock basic")
```
Comment (adityadani) [inline test/kv.go:923]: Do you need these print statements?

**Review by adityadani — CHANGES_REQUESTED** [review]
Overall looks good. A few minor comments

File: etcd/v3/kv_etcd.go (hunk 1 above)
Comment (vlaxman-px) [reply etcd/v3/kv_etcd.go:737]: Sure I'll change that

File: test/kv.go (hunk 2 above)
Comment (vlaxman-px) [reply test/kv.go:911]: Ok, it was just to check if it's working properly, I'll remove that

File: test/kv.go (hunk 3 above)
Comment (vlaxman-px) [reply test/kv.go:923]: Nope, I'll remove that

**Review by adityadani — APPROVED** [review]

File: etcd/v3/kv_etcd.go [hunk 4]
```diff
@@ -733,7 +733,10 @@ func (et *etcdKV) LockWithTimeout(
 			}
 		}
 		if err != nil && time.Since(startTime) > lockTryDuration {
-			return nil, err
+			currLockerTag := ec.LockerIDInfo{LockerID: ""}
+			if _, errGet := et.GetVal(key, &currLockerTag); errGet == nil {
+				return nil, fmt.Errorf("failed to take a lock on %v: lock taken by: %v ", key, currLockerTag.LockerID)
```
Comment (adityadani) [inline etcd/v3/kv_etcd.go:738]: We should still return an error here if errGet != nil

Comment (adityadani) [reply etcd/v3/kv_etcd.go:738]:
```suggestion
			if _, errGet := et.GetVal(key, &currLockerTag); errGet == nil {
				return nil, fmt.Errorf("failed to take a lock on %v: lock taken by: %v ", key, currLockerTag.LockerID)
			}
			return nil, fmt.Errorf("failed to take a lock on %v", key)
```

Comment (vlaxman-px) [reply etcd/v3/kv_etcd.go:738]: Okay, I'll raise a new PR for this one because this one has been closed

**Pull request merged**